pkg/reconciler/revision/resources/deploy.go

```diff
@@ -145,8 +145,10 @@ func BuildUserContainers(rev *v1.Revision) []corev1.Container {
 		} else {
 			container = makeContainer(*rev.Spec.PodSpec.Containers[i].DeepCopy(), rev)
 		}
+		// The below logic usually mislead because of the operation on status during creation
+		// But as all images will be resolved to digests before calling this function
+		// so rev.Status.ContainerStatuses can be used to override the container images.
```

Comment:
Added a bit about the dry run case, what do you think?

```suggestion
		// The below logic is safe because the image digests in Status.ContainerStatus will have been resolved
		// before this method is called. We check for an empty array here because the method can also be 
		// called during DryRun, where ContainerStatuses will not yet have been resolved.
```